python/lsst/pipe/tasks/selectImages.py

```diff
@@ -524,8 +524,16 @@ def run(self, visitSummaries, skyMap, dataId):
             output = sortedVisits
         else:
             output = sortedVisits[:self.config.nVisitsMax]
-        self.log.info("%d images selected with FWHM range of %f--%f arcseconds",
-                      len(output), fwhmSizes[visits.index(output[0])], fwhmSizes[visits.index(output[-1])])
+
+        if len(output) == 0:
+            self.log.info("All images rejected in BestSeeingSelectVisitsTask.")
+        else:
+            self.log.info(
+                "%d images selected with FWHM range of %f--%f arcseconds",
+                len(output),
+                fwhmSizes[visits.index(output[0])],
+                fwhmSizes[visits.index(output[-1])],
+            )
```

Comment:
Ok, so this works because `goodVisits` ends up as an empty `dict`, so `butlerQC.put()` is effectively a no-op.  When I encountered a similar situation on DM-41512, I asked about this behavior and whether it's better to explicitly add a `NoWorkFound` (in my case I was sending `None` to `butlerQC.put()`...the consensus was yes, hence: https://github.com/lsst/pipe_tasks/blob/main/python/lsst/pipe/tasks/finalizeCharacterization.py#L375-L376